pyccel/parser/semantic.py

```diff
@@ -944,6 +944,12 @@ def _extract_indexed_from_var(self, var, args, name):
                 bounding_box=(self._current_fst_node.lineno, self._current_fst_node.col_offset),
                 severity='error', blocker=self.blocking)
 
+        for arg in var[args]._indices:
+            if not isinstance(arg, (Slice, LiteralInteger)) and not \
+                (isinstance(arg, Variable) and isinstance(arg.dtype, (NativeBool, NativeInteger))):
```

Comment:
Why does the index need to be a `Variable`. Can we not do things like:
`a[i+j]` ? As long as we have the right dtype we should be fine